src/optimization/ControlFlow.cpp

```diff
@@ -1143,8 +1148,37 @@ void optimizations::removeConstantLoadInLoops(const Module& module, Method& meth
                     auto out = loadInst->getOutput().value();
                     if(out.valueType == ValueType::LOCAL)
                     {
-                        auto rootInst = root->key->findPredecessor()->key->end().previousInBlock();
-                        rootInst.emplace(it.release());
+                        if(insertedBlock != nullptr)
+                        {
+                            insertedBlock->end().emplace(it.release());
+                        }
+                        else
+                        {
+                            auto targetBlock = root->key->findPredecessor();
+                            if(targetBlock != nullptr)
+                            {
+                                auto targetInst = targetBlock->key->end().previousInBlock();
```

Comment:
why do you insert the instruction one-before-then-end and not at the end (which is done in the case above)?

Reply:
I misunderstood a specification of `InstructionWalker::emplace`. I will fix it.